core/runtime/src/main/java/io/quarkus/runtime/StartupContext.java

```diff
@@ -78,6 +78,8 @@ public boolean isLastValueSet() {
     public void close() {
         runAllAndClear(shutdownTasks);
         runAllAndClear(lastShutdownTasks);
+        values.clear();
```

Comment:
Out of curiosity - does anyone happen to know what exactly are these values used for? The commit message is `Add some stuff` and that does not really help. It wouldn't hurt to improve the javadoc here...

CC @stuartwdouglas @geoand

Reply:
@mkouba see https://github.com/quarkusio/quarkus/blob/ad963893983cc8445cc43415ff283bd4e8efe898/core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java#L542 and https://github.com/quarkusio/quarkus/blob/ad963893983cc8445cc43415ff283bd4e8efe898/core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java#L729

Reply:
Hm, so basically if a recorder method returns a proxy we store it in the `StartupContext` and then when it's used as an argument of another recorder method we obtain the instance from the `StartupContext` at runtime.

Reply:
I wonder if it's the only use case for `values`? If so the `values` should be probably renamed to something like `recordedProxies` or something like that.

Reply:
> I wonder if it's the only use case for values

it appears so, unless I missed something

Reply:
I will merge this but I wouldn't mind a follow-up.

Also, if this `StartupContext` is only used when starting the application, shouldn't we drop it once we have run the runtime bytecode recording?

Reply:
> Also, if this StartupContext is only used when starting the application, shouldn't we drop it once we have run the runtime bytecode recording?

We can't because we use it to hold the shutdown tasks...

I will try to rename the field/methods and we'll see if it breaks something or not ;-)

Reply:
But the shutdown tasks are very specific? Couldn't we isolate them and avoid keeping the rest around?

Now I'm not sure exactly if it would improve things. Typically, maybe in the end, most of them are referenced elsewhere anyway.

Reply:
I'm not sure but I'd rather be careful here. I mean there's a lot of complex (and often undocumented) logic behind so even a small change can break a lot of stuff...